### ISSUE TITLE
[Bug]: drop table hung, in pessimistic mode.

### ISSUE THREAD

**Issue by fengttt** [matrixorigin/matrixone#10488]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

I reproed this with my little toy mojo. Feel free to do you own repro from the code but i kind of starting to like my little tool.

https://github.com/matrixorigin/mojo/tree/main/go

https://github.com/matrixorigin/mojo/blob/main/go/pkg/repro/moc956.go

!repro moc-956
This will drop 200 tables, then loop create table, insert select result, result+1, result+2 from generate_series(1, 10000) tmpt;. Note that this will insert 10K rows per table in about 0.1 seconds.

Now, !repro moc-956 -b single -s, or !repro moc-956 -b prepare -s, this will insert using either single insert, or a prepared statement. Only 1K rows, 23 second. 1 thousand times slower! -s will only insert a single table.

Now !repro moc-956 -b single -t begin -s. This will insert all in one table in a big transaction. 15 seconds.

Now, !repro moc-956 -b single -t begin, this will try to insert into 200 tables, 15 seconds each. That means it will take an hour. Kill it at the 2 or 3 table.

Restart mojo. Run !repro moc-956 again. It will try to drop tables, but hung at the table that we issued kill. Seems we did not release ALL locks cleanly.

### Expected Behavior

Found this bug when I try to repro moc-956

**Comment by fengttt** [comment]
https://github.com/matrixorigin/MO-Cloud/issues/956

**Comment by zhangxu19830126** [comment]
not working on this

**Comment by zhangxu19830126** [comment]
not working on this

**Comment by zhangxu19830126** [comment]
not working on this

**Comment by zhangxu19830126** [comment]
not working on this

**Comment by zhangxu19830126** [comment]
not working on this

**Comment by zhangxu19830126** [comment]
maybe fixed by #10816. i cannot repro.
